packages/gestalt/src/TextField.js

```diff
@@ -149,7 +149,12 @@ const TextFieldWithForwardRef: React$AbstractComponent<
               {tag}
             </Box>
           ))}
-          {inputElement}
+          <Box flex="grow" marginEnd={2} maxWidth="100%" position="relative">
+            <div aria-hidden className={styles.measure}>
+              {value}
+            </div>
```

Comment:
Also probably worth adding a comment here to explain this div's purpose for future users (the details in this PR description are nice and clear)